pkg/execution/worker.go

```diff
@@ -38,16 +38,13 @@ func (w *FQLWorker) IsRunning() bool {
 
 func (w *FQLWorker) Process() ([]byte, error) {
 	w.mu.Lock()
-	isRunning := w.cancel != nil
-	w.mu.Unlock()
 
-	if isRunning {
+	if w.cancel != nil {
 		return nil, ErrWorkerAlreadyRunning
```

Comment:
If you exit here, next function call will cause a deadlock.
Put ```w.mu.Unlock()``` please.

Reply:
@ziflex 👍